### DISCUSSION TITLE
Menubutton title based on selected value

### DISCUSSION THREAD

**Discussion by youstanzr** [israel-dryer/ttkbootstrap#241] [Q&A] [answered]
I am trying to use menubutton to select a value from a list. I have created my menu of radiobutton but I was unable to show the text of the menu button as text and filter that out from the choices. So, is that possible to be done with the current implementation? \is there even a hacky way of doing it?

Also it would be helpful if there was a value_did_change function for when radiobutton choice is changed but I was unable to find that functionality

**Comment by israel-dryer** [comment]
> I am trying to use menubutton to select a value from a list. I have created my menu of radiobutton but I was unable to show the text of the menu button as text and filter that out from the choices. So, is that possible to be done with the current implementation? \is there even a hacky way of doing it?
> 
> Also it would be helpful if there was a value_did_change function for when radiobutton choice is changed but I was unable to find that functionality

Yes, I believe you can do what you want with the current functionality.

Would you mind posting a minimal example so that I can see what you are trying to do and what you have done so far. It will be easier to start with this than to guess what you mean.

**Comment by rdbende** [comment, marked as answer]
> Menubutton title based on selected value

Have you tried the `OptionMenu` widget? It does exactly that.

**Reply by youstanzr** [reply]
Is there like a sample code of this? I did not see this in the documentation nor the gallery of examples

**Reply by youstanzr** [reply]
Got it by searching through repo. Thanks!

	menu = ttk.Menu(self)
	menu_var = ttk.StringVar(value=list(df.columns)[0])
	filterColMenu = ttk.OptionMenu(self, menu_var, menu_var.get(), *list(df.columns), bootstyle='secondary')
	filterColMenu.grid(row=1, column=0, padx=10, pady=5, sticky='w')

I wish that this was in the documentation, it would have been extremely helpful. This is the second class that I try to handle myself and I later find that it is already available (the other one is tableview but I ended up keeping my own wrapper cause I was already done). Such an amazing library that makes things so much easier.

**Comment by youstanzr** [comment]
Here is some of the code I am trying to do

	menu = ttk.Menu(self)
	menu_var = ttk.IntVar(value=0)
	for i, t in enumerate(list(df.columns)):
		menu.add_radiobutton(label=t, value=i, variable=menu_var)
	filterColMenu = ttk.Menubutton(self, text=list(df.columns)[0], menu=menu, bootstyle='secondary')

Now the issue that when I change the selection the text does not change and the choice is not removed from the options
